main.go

```diff
@@ -41,6 +41,13 @@ func getMetrics(zkOpts zkOptions) map[string]string {
 		if err != nil {
 			log.Printf("warning: cannot connect to %s: %v", h.Unresolved, err)
 			metrics[fmt.Sprintf("zk_up{%s}", hostLabel)] = "0"
+			newTcp, _ := net.ResolveTCPAddr("tcp", h.Unresolved)
+			if newTcp.IP.String() != h.TCPAddr.IP.String() {
+				log.Printf("INFO: resolve %s to new ip: %s ( old ip was: %s )", h.Unresolved, newTcp, h.TCPAddr)
+				zkOpts.Hosts[i].TCPAddr = newTcp
+			}else{
+				errFatal(err)
```

Comment:
Fatal may be too brutal here, we're checking `Dial` error and log warning above (L42).
This should be enough since we want exporter to continue checking other zk hosts in the list.